.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,178 @@
+# Adapted from NumPy https://github.com/numpy/numpy/pull/25894
+# https://github.com/numpy/numpy/blob/d2d2c25fa81b47810f5cbd85ea6485eb3a3ffec3/.github/workflows/emscripten.yml
+#
+
+name: Pyodide
+
+on: [push, pull_request]
+
+env:
+  FORCE_COLOR: 3
+  DURATIONS_CACHE_NAME: sympy_pyodide_test_durations_gha_cache
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read # to fetch code (actions/checkout)
+
+jobs:
+  pyodide-test:
+    name: Test group ${{ matrix.group }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        splits: [4] # keep in sync with number of groups below
+        group: [1, 2, 3, 4]
+    env:
+      PYODIDE_VERSION: 0.27.0a2
+      # PYTHON_VERSION and EMSCRIPTEN_VERSION are determined by PYODIDE_VERSION.
+      # The appropriate versions can be found in the Pyodide repodata.json
+      # "info" field, or in Makefile.envs:
+      # https://github.com/pyodide/pyodide/blob/main/Makefile.envs#L2
+      PYTHON_VERSION: 3.12.1
+      EMSCRIPTEN_VERSION: 3.1.58
+      NODE_VERSION: 20
+    steps:
+      - name: Checkout SymPy
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
```

Comment:
We would not normally pin these so tightly but rather just use `@v4` so it doesn't need to be updated so often.

Reply:
Yes, makes sense to me. Addressed in 0040f74652b61482576d416ecd8b7ab52adc9b55.